pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -112,33 +112,35 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 	// TODO (andreyvelich): Add validation to check that TrainJob doesn't have "PET_" envs.
 	// TODO (andreyvelich): We should validate that envs from different plugins don't conflict with each other.
 	// Ref: https://github.com/kubeflow/trainer/pull/2308#discussion_r1823229940
+	var trainerContainer *runtime.Container
 	if trainJob.Spec.Trainer != nil {
-		info.Trainer.Env = apply.EnvVars(trainJob.Spec.Trainer.Env...)
+		if trainerContainer = info.FindContainerByPodSetContainerName(constants.JobTrainerNode, constants.ContainerTrainer); trainerContainer != nil {
+			apply.UpsertEnvVars(&trainerContainer.Env, apply.EnvVars(trainJob.Spec.Trainer.Env...)...)
+		}
+	}
+	if trainerContainer != nil {
```

Comment:
We have this check to make sure it won't fail in case of validation webhook disabled ?

Reply:
Yes, that's right. And want to avoid null pointer reference.